### PR TITLE
[functorch][reland] vmap: bitwise operators

### LINKED ISSUE #1069: Performance drop because of not yet implemented batching rule.
Hello, @zou3519 , @samdow 

I get the following warning when i try to compute jacobians of my model. 

UserWarning: There is a performance drop because we have not yet implemented the batching rule for aten::bitwise_xor_.Tensor. Please file us an issue on GitHub so that we can prioritize its implementation. (Triggered internally at /opt/conda/conda-bld/pytorch_1666642991888/work/aten/src/ATen/functorch/BatchedFallback.cpp:82.)

The above error is triggered by the code below.

`def spatial_hash(coords: List[Tensor]) -> Tensor:
    PRIMES = (1, 2654435761, 805459861, 3674653429)
    #assert len(coords) <= len(PRIMES), "Add more PRIMES!"
    if len(coords) == 1:
        i =  (coords[0] ^ PRIMES[1])# torch.logical_xor(coords[0], torch.tensor(PRIMES[1]).cuda()) # (coords[0] ^ PRIMES[1])
    else:
        i = coords[0] ^ PRIMES[0] # torch.logical_xor(coords[0], torch.tensor(PRIMES[0]).cuda()) # coords[0] ^ PRIMES[0]
        for c, p in zip(coords[1:], torch.tensor(PRIMES[1:]).cuda()):
            i  ^= c * p # torch.logical_xor(i, c * p) # ^= c * p
    return i`

Are there plans to implemented batching for the above operations in the near future ? 

Thank you

### PR DESCRIPTION
**Pull request by khushi-411**
Previous PR: #91971

Fixes: https://github.com/pytorch/functorch/issues/1069

cc: @Chillee @kshitij12345

### CODE REVIEW

**Comment by khushi-411** [comment]
Hi, Gentle Ping @Chillee!

**Review by Chillee — APPROVED** [review]

**Comment by Chillee** [comment]
Looks like a reland so stamping, but would be good to have a comment on what changed with the previous PR.

File: test/functorch/test_vmap.py [hunk 1]
```diff
@@ -3730,14 +3730,17 @@ def test_vmap_exhaustive(self, device, dtype, op):
         xfail('linalg.lu', ''),
         skip('linalg.ldl_solve', ''),
         skip('_softmax_backward_data'),
+        # AssertionError: Tensor-likes are not equal!
+        # Issue: https://github.com/pytorch/pytorch/issues/70904
+        xfail('bitwise_left_shift', device_type='cpu'),
+        decorate('bitwise_right_shift', device_type='cpu',
+                 decorator=expectedFailureIf(not (IS_MACOS and IS_X86))),
```
Comment (khushi-411) [inline test/functorch/test_vmap.py:3737]: Thanks for reviewing, @Chillee!

As per [CI failure](https://hud.pytorch.org/pytorch/pytorch/commit/e54f7b3edde356c97c99706942f4b32a5a5ba475), I see the `bitwise_right_shift` test passes with the `macOS` platform in `x86_64` architecture. while testing cross-platform. Earlier I xfailed with the CPU device type.

Please correct me if I'm wrong. Thanks for the question! :)

**Pull request closed without merging**